react-front-end/tsrc/search/SearchPage.tsx

```diff
@@ -197,6 +200,9 @@ const SearchPage = ({ updateTemplate, advancedSearchId }: SearchPageProps) => {
   const [advancedSearches, setAdvancedSearches] = useState<
     OEQ.Common.BaseEntitySummary[]
   >([]);
+  const [wizardDefinition, setWizardDefinition] = useState<
```

Comment:
Just a quick one while you fix the tests though... Would make sense to go with a more specific name here:

```suggestion
  const [advancedSearchDefinition, setAdvancedSearchDefinition] = useState<
```

Mind you though, we've quickly accumulated a bit of state relating to advanced search mode, should we merge it - maybe even another reducer I wonder.

Bits that come to mind:

- this definition
- show/hide the panel
- maybe the id
- and soon the values or query...

hmmm, what do you think?

Reply:
> The new tests for the Advanced Search Panel displaying logic are failing @PenghaiZhang , and seeing they're pretty critical to this I better wait until you fix to review.

Ah I forgot to mock the new function. No problems. Fixed.

Reply:
At the moment we have two. How about we merge them into a object first ?

Reply:
Sounds good, we can start there and see how it goes when we add the others.

But keeping in mind that one of the key purposes of a reducer is to manage multiple pieces of related state. And where this was already jumping out at me was when you were checking for both show advanced search panel and whether there was a wizard definition.

Reply:
I have implemented a reducer for this.